asciidoc/courses/genai-mcp-build-custom-tools-python/modules/1-getting-started/lessons/3c-create-first-server/lesson.adoc

```diff
@@ -0,0 +1,187 @@
+= Create Your First MCP Server 
+:type: challenge
+:order: 3
+
+Now for a challenge!  So far you've learned about the MCP Python SDK and FastMCP, and set up your development environment.
+
+Now it's time to put that knowledge into practice by creating your own MCP server from scratch.
+
+Your challenge is to create a simple MCP server that demonstrates the core concepts you've learned, then run and test it using the tools you've set up.
+
+[NOTE]
+.GitHub Codespace
+====
+If you haven't completed the setup lesson yet, go back to link:../2-setup/[Setting Up Your Development Environment^] to configure your Codespace and environment variables.
+
+You can keep the Codespace open while completing the course. 
+You can open your existing codespaces at link:https://github.com/codespaces[github.com/codespaces^].
+====
+
+
+== Challenge Goals
+
+To complete this challenge, you will need to:
+
+1. Create a new MCP server using the MCP Python SDK.
+2. Create a tool that counts the number of times a letter appears in a word.
+3. Configure VS Code to use the MCP server with stdio.
+4. Test the server using the Chat window in Agent mode.
+
+== Step-by-Step Instructions
+
+=== Step 1: Set up your project directory
+
+First, create a new directory for your MCP server:
+
+[source,bash]
+----
+mkdir strawberry
+cd strawberry
+----
+
+=== Step 2: Initialize the project with uv
+
+Initialize a new Python project using `uv`:
+
+[source,bash]
+----
+uv init
+----
+
+This will create a basic project structure with a `pyproject.toml` file and a `main.py` file that you will use to build your MCP server.
+
+=== Step 3: Add the MCP dependency
+
+Add the MCP Python SDK to your project:
+
+[source,bash]
+----
+uv add "mcp[cli]"
+----
+
+=== Step 4: Create your MCP server
+
+
+Open `main.py` in your editor and add the following code to create a simple MCP server:
```

Comment:
`main.py` isnt in the repo. I was a little confused about where to put main.py and had to refer to the solution.

```suggestion
Create a new file, `strawberry\main.py`, and add the following code to create a simple MCP server:
```